### DISCUSSION TITLE
L298 driver

### DISCUSSION THREAD

**Discussion by Geoide** [Moddable-OpenSource/moddable#915] [Porting]
Hello.

Is there some driver to manage the L298 motor controller?

Thanks.

**Comment by phoddie** [comment]
It looks like the [`j5e/motor`](https://j5e.dev/modules/motor) class of J5e may support L298. I'm not familiar enough to motors to be sure, but the [implementation](https://github.com/dtex/j5e/blob/main/lib/motor/index.js) appears to support the PWM and direction pins that the L298 has. @dtex, can you help here?

FYI – if you aren't familiar with J5e, it is a very nice robotics library for JavaScript built on the [ECMA-419 standard](https://419.ecma-international.org), the ECMAScript Embedded Systems API Specification. J5e runs on the Moddable SDK's ECMA-419 implementation.

**Comment by dtex** [comment]
Yes, it does support the L298, but the docs are lacking here because I want to do more testing. I can give you some sample code, but not all L298 based motor controllers are equal. Can you tell me what input pins your board has, or even better, send a link to a spec sheet?

**Comment by AlexLucardGoodman** [comment]
Good afternoon!
Im trying to get the L298n motor driver to work with J5e/motor class on an ESP32 but haven't been able to yet. Theres very little documentation and a few short examples which dont really help much. Any suggestions on how to get this done? This is the last test i wrote:

```js
import Motor from "j5e/motor";
import PWM from "pins/pwm";

export async function createMotor(){

	let motorAPinEnable = 1;
	let motorAPin1 = 23;
	let motorAPin2 = 22;

        const speedIO = new device.io.PWM({
	   pin: motorAPinEnable
	});

        const motor = await new Motor([speedIO,motorAPin1,motorAPin2]);

        motor.enable();
	motor.start();
	motor.forward(150);
}
```
Nothing happens haha! Thanks in advance

**Reply by phoddie** [reply]
@dtex, any suggestions for diagnosing this?
